### PR TITLE
:zap: Support webpack 5

### PR DESCRIPTION
**Pull request by RDIL**
Signed-off-by: Reece Dunham <me@rdil.rocks>

## Summary
- Add support for webpack 5
- Drop support for webpack 3, its quite old and I don't know anybody that uses it anymore
- Other minor fixups

I've also already published a package based on my fork, so if you don't want to merge this that is fine.

## Why
webpack 5 is the next big upgrade for facebook/docusaurus, so this is needed.

## Checklist

- [X] Your code builds clean without any `errors` or `warnings`
- [X] You are using `approved terminology`
- [X] You have added `unit tests`, if apply.

## Emojis for categorizing pull requests:

⚡️ New feature (`:zap:`)
🐛 Bug fix (`:bug:`)  
🔥 P0 fix (`:fire:`)  
✅ Tests (`:white_check_mark:`)  
🚀 Performance improvements (`:rocket:`)  
🖍 CSS / Styling (`:crayon:`)  
♿ Accessibility (`:wheelchair:`)  
🌐 Internationalization (`:globe_with_meridians:`)  
📖 Documentation (`:book:`)  
🏗 Infrastructure / Tooling / Builds / CI (`:building_construction:`)  
⏪ Reverting a previous change (`:rewind:`)  
♻️ Refactoring (like moving around code w/o any changes) (`:recycle:`)  
🚮 Deleting code (`:put_litter_in_its_place:`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "stats": "NODE_ENV=development webpack --profile --json > compilation-stats.json"
   },
   "engines": {
-    "node": ">=4.2.4"
+    "node": ">=10.13.0"
```
Comment (RDIL) [inline package.json:39]: To comply with webpack's required version

**Comment by themgoncalves** [comment]
@RDIL thanks for the merge request, I will be handling it over the weekend

**Review by themgoncalves — APPROVED** [review]
LGTM.

Will be releasing a `major` version, as the drop of Webpack v3 support is a breaking change.

**Comment by themgoncalves** [comment]
@RDIL `v1.0.0` released.

**Pull request merged**